app/api/events.py

```diff
@@ -115,78 +126,214 @@ def update_event(request: Request, id:str, eventUpdate: EventUpdate, AccessToken
 
     return Response(status_code=200)
 
+
 @router.get('/{id}', dependencies=[Depends(validate_uuid)])
-def get_event_by_id(request: Request, id:str):
+def get_event_by_id(request: Request, id: str):
     db = get_database(request)
 
     event = get_event_or_404(db, id)
     return EventDB.parse_obj(event)
 
+# TODO add admin validation in get_event_participants
+# Create a new endpoint to return number of joined participants.
+
+
 @router.get('/{id}/participants', dependencies=[Depends(validate_uuid)])
-def get_event_participants(request: Request, id:str):
+def get_event_participants(request: Request, id: str, token: AccessTokenPayload = Depends(authorize)):
     db = get_database(request)
 
     event = get_event_or_404(db, id)
+    if token.role == "admin":
+        return [{'id': p['id'], 'name': p['realName'],
+                 'email': p['email'], 'classof': p['classof'],
+                 'phone':p['phone'], 'role': p['role'],
+                 'food': p['food'], 'transport': p['transport'],
+                 'dietaryRestrictions':p['dietaryRestrictions'],
+                 'submitDate': p['submitDate']} for p in event['participants']]
+
+    return [{'id': p['id'], 'name': p['realName']} for p in event['participants']]
 
-    return [{'id' : p['id'], 'name': p['realName']} for p in event['participants']]
 
 # 400 if already joined?
 @router.post('/{id}/join', dependencies=[Depends(validate_uuid)])
-def join_event(request: Request, id:str, token: AccessTokenPayload = Depends(authorize)):
+def join_event(request: Request, id: str, payload: JoinEventPayload, token: AccessTokenPayload = Depends(authorize)):
     db = get_database(request)
-    
     event = get_event_or_404(db, id)
 
     member = db.members.find_one({'id': token.user_id})
 
     if not member:
         raise HTTPException(400, "User could not be found")
 
-    db.events.update_one({'eid' : event['eid']}, { "$addToSet": { "participants": member } })
+    participantData = (
+        token.user_id, member['realName'],
+        member['email'], member['classof'],
+        member['phone'], member['role'],
+        payload.food,
+        payload.transport,
+        payload.dietaryRestrictions,
+        datetime.now()
+    )
+
+    fields = Participant.__fields__.keys()
```

Comment:
```python3
    participantData = {
        'food': payload.food,
        'transport': payload.transport,
        'dietaryRestrictions': payload.dietaryRestrictions,
        'submitDate': datetime.now()
    }

    new_fields = { **member, **participantData }
    participant = Participant.parse_obj(new_fields)
```
i think something like this would be more suitable